api/src/main/java/eu/coatrack/api/HashedApiKey.java

```diff
@@ -0,0 +1,35 @@
+package eu.coatrack.api;
+
+/*-
+ * #%L
+ * coatrack-api
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import lombok.NoArgsConstructor;
+
+import java.util.Date;
+
+@NoArgsConstructor
+public class HashedApiKey {
```

Comment:
Just brainstorming: should this maybe be a subclass of `ApiKey`? But maybe this would lead to problems as this is just a DTO, while `ApiKey` is an entity. To be discussed.

Reply:
The problem of a subclass would be that ApiKey contains an apiKeyValue field and HashedApiKey contains a hashedApiKeyValue field. If inheritance would be applied, hashedApiKeyValue would contain a redundant apiKeyValue field. In addition to that, not all fields in of ApiKey are required within the Gateway which would also be redundant in case of using inheritance.